src/compiler/binder.ts

```diff
@@ -3340,6 +3341,11 @@ function createBinder(): (file: SourceFile, options: CompilerOptions) => void {
                 }
             });
         }
+        if (isPrototypeProperty && namespaceSymbol && namespaceSymbol.valueDeclaration && isAssignmentDeclaration(namespaceSymbol.valueDeclaration)) {
+            //A prototype property assignment to something that looks class-like shouldn't actually count if the class-likeness came from an assignment declaration.
+            //The checker requires an actual class/constructor function to attach the prototype property to.

```

Comment:
Just saw that you also left a comment - if you want to stick to this comment, try to shorten the line and add a space after each `//`

```suggestion
            // A prototype property assignment to something that looks class-like
            // shouldn't actually count if the class-likeness came from an assignment declaration.
            // The checker requires an actual class/constructor function to attach the prototype property to.

```